components/compositing/compositor.rs

```diff
@@ -368,23 +360,6 @@ impl webrender_traits::RenderDispatcher for CompositorThreadDispatcher {
 impl<Window: WindowMethods> IOCompositor<Window> {
     fn new(window: Rc<Window>, state: InitialCompositorState)
            -> IOCompositor<Window> {
-        // Register this thread as a memory reporter, via its own channel.
-        let (reporter_sender, reporter_receiver) = ipc::channel()
-            .expect("Compositor reporter chan");
-        let compositor_proxy_for_memory_reporter = state.sender.clone_compositor_proxy();
-        ROUTER.add_route(reporter_receiver.to_opaque(), box move |reporter_request| {
-            match reporter_request.to::<ReporterRequest>() {
-                Err(e) => error!("Cast to ReporterRequest failed ({}).", e),
-                Ok(reporter_request) => {
-                    let msg = Msg::CollectMemoryReports(reporter_request.reports_channel);
```

Comment:
I suspect the code that receives and processes this message can be removed as well. Probably the message variant from the enum, too.